File: Week 9/LargestNumber/uno.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool cmp(string a, string b) {
	return a + b > b + a;
}
string solution(vector<int> numbers) {
	string answer;
	vector<string> temp;

	for (int i : numbers) {
		temp.push_back(to_string(i));
	}

	sort(temp.begin(), temp.end(), cmp);

	if (temp[0] == "0") {
		answer += '0';
		return answer;
	}
	for (string s : temp) {
		answer += s;
	}

	return answer;
}

File: Week 9/LargestNumber/uno.py
import functools


def cmp(a, b):
    t1 = a + b
    t2 = b + a

    return (int(t1) > int(t2)) - (int(t1) < int(t2))


def solution(numbers):
    answer = ""
    temp = [str(x) for x in numbers]
    temp = sorted(temp, key=functools.cmp_to_key(cmp), reverse=True)

    for s in temp:
        answer += s

    return answer


print(solution([6, 10, 2]))
